experimental/ComplexReflectionGroups/src/ComplexReflectionGroupType.jl

```diff
@@ -0,0 +1,678 @@
+                return false
+            end
+        end
+    end
+
+    return true
+end
+
+function is_weyl_group(G::ComplexReflectionGroupType)
+    return is_rational(G)
+end
+
+function coxeter_number(G::ComplexReflectionGroupType)
+
+    return div(num_reflections(G) + num_hyperplanes(G), rank(G))
+
+end
+
+# function is_shephard_group(G::ComplexReflectionGroupType)
+
+#     # See Lehrer & Taylor (2009), Tables 6.1, 6.2, 6.3
+#     excshep = [ 4, 5, 6, 8, 9, 10, 14, 16, 17, 18, 20, 21 ]
+
+#     for t in G.type
+#         if length(t) == 1
+#             if !(t[1] in excshep)
+#                 return false
+#             end
+#         else
+#             m = t[1]
+#             p = t[2]
+#             n = t[3]
+#             if p != 1 
+#                 return false
+#             end
+#         end
+#     end
+
+#     return true
+
+# end
+
+# function is_duality_group(G::ComplexReflectionGroupType)
+
+# end
+
+function is_spetsial(G::ComplexReflectionGroupType)
+    
+    # See Achar (2009)
+    excspet = [ 4, 6, 8, 14, 23, 24, 25, 26, 27, 28, 29, 30, 32, 33, 34, 35, 36, 37 ]
+
+    for t in G.type
+        if isa(t, Int)
+            if !(t in excspet)
+                return false
+            end
+        else
+            m = t[1]
+            p = t[2]
+            n = t[3]
+            if !( (p == 1) || (m == p) )
+                return false
+            end
+        end
+    end
+
+    return true
+end
+
+function num_reflection_classes(G::ComplexReflectionGroupType)
+
+    N = ZZ(0)
+
+    # Grabbed from Magma
+    excnum = [ 2, 4, 3, 5, 3, 4, 5, 6, 1, 2, 3, 4, 4, 5, 6, 7, 2, 3, 1, 1, 1, 2, 3, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1 ]
+
+    for t in G.type
+        if isa(t, Int)
+            i = t
+            N += ZZ(excnum[i-3]) #index shift by 3 because exceptionals start with G4
+        else
+            (m,p,n) = t
+            
+            # See Thiel (2014), Theorem 15.27
+            if n > 2 || (n == 2 && is_odd(p))
+                N += div(ZZ(m),ZZ(p))
+            else
+                N += div(ZZ(m),ZZ(p)) + 1
+            end
+        end
+    end
+
+    return N
+
+end
+
+function center(G::ComplexReflectionGroupType)
+
+    orders = ZZRingElem[]
+
+    # The center of an irreducible group is always cyclic of order equal to the gcd
+    # of the degrees.
+    # See Thiel (2014), Proposition 13.32
+    for C in components(G)
+        append!(orders,gcd(degrees(C)))
+    end
+
+    return abelian_group(orders)
+
+end
+
+function field_of_definition(G::ComplexReflectionGroupType)
+    # Reference is Geck & Malle (2006)
+
+    fields = Field[]
+
+    for t in G.type
+        if isa(t, Int)
+            i = t
+            if i == 4
+                K,z = cyclotomic_field(3)
+            elseif i == 5
+                K,z = cyclotomic_field(3)
+            elseif i == 6
+                K,z = cyclotomic_field(12)
+            elseif i == 7
+                K,z = K,z = cyclotomic_field(12)
```

Comment:
```suggestion
                K,z = cyclotomic_field(12)
```
Also some times more in the following lines.